### PR TITLE
mon/OSDMonitor: Improve encode_pending operation

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

**Comment by shinobu-x** [comment]
test this please

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -1230,6 +1033,180 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
...
+    for (auto &pool : tmp.get_pools()) {
+      auto p = pool.first;
+      if (tmp.get_pg_pool(p)->has_flag(pg_pool_t::FLAG_NEARFULL) &&
+          nearfull_pool_ids.empty()) {
+        dout(10) << __func__ << " clearing pool '" << tmp.pool_name[p]
+                 << "'s nearfull flag" << dendl;
+        if (pending_inc.new_pools.count(p) == 0) {
+          pending_inc.new_pools[p] = pool.second;
+        }
+        pending_inc.new_pools[p].flags &= ~pg_pool_t::FLAG_BACKFILLFULL;
```
Comment (jecluis) [inline src/mon/OSDMonitor.cc]: what happened here? This should have been `FLAG_NEARFULL`, and instead is `FLAG_BACKFILLFULL`, whereas the `if case` for backfillfull is simply gone. I don't think those two things are the same, so this feels a log like a copy-paste error.

**Comment by shinobu-x** [comment]
@jecluis 
Thanks for your review which is appreciated :)
However since its status is DNM, what I would like to achieve has not been completed...

**Comment by jecluis** [comment]
@shinobu-x alright, sorry - didn't realize there was more to come ;) feel free to ping me when the branch is up for review then!

**Comment by shinobu-x** [comment]
test this please

**Comment by batrick** [comment]
@shinobu-x status on this PR?

**Pull request closed without merging**